apiserver/facades/client/client/interface.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package client
+
+//go:generate mockgen -package mocks -destination mocks/cache.go github.com/juju/juju/apiserver/facades/client/client ModelCache,ModelCacheBranch
+
+// ModelCache represents point of use methods from the cache
+// model.
+type ModelCache interface {
```

Comment:
While I like indirecting services like this, for a suite that embeds `JujuConnSuite` I am not sure we should expose patchable methods to get at them.

Since the suite ensures that its cache is populated from the DB, all we really need to do is sync after making DB changes and before evaluating the tests, like I did [here](https://github.com/juju/juju/pull/10378/commits/efd60753e53bb6bd2c3424cffe6b83f0c4d9cd3d).

Reply:
StartSync() and WaitForModelWatchersIdle() did the trick to get the tests working without the mocks.